optuna/samplers/_tpe/sampler.py

```diff
@@ -477,59 +476,75 @@ def _sample(
             self._constraints_func is not None,
         )
 
-        below = self._get_internal_repr(below_trials, search_space)
-        above = self._get_internal_repr(above_trials, search_space)
+        mpe_below = self._build_parzen_estimator(
+            study, search_space, below_trials, handle_below=True
+        )
+        mpe_above = self._build_parzen_estimator(study, search_space, above_trials)
+
+        samples_below = mpe_below.sample(self._rng.rng, self._n_ei_candidates)
+        acq_fn_vals = self._compute_acquisition_function(samples_below, mpe_below, mpe_above)
+        ret = TPESampler._compare(samples_below, acq_fn_vals)
+
+        for param_name, dist in search_space.items():
+            ret[param_name] = dist.to_external_repr(ret[param_name])
+
+        return ret
 
-        # We then sample by maximizing log likelihood ratio.
-        if study._is_multi_objective():
+    def _build_parzen_estimator(
+        self,
+        study: Study,
+        search_space: dict[str, BaseDistribution],
+        trials: list[FrozenTrial],
+        handle_below: bool = False,
```

Comment:
I feel using the default argument, which could cause bugs, here is unnecessary. How about avoiding removing the default argument?